x-pack/plugin/relevance-search/src/main/java/org/elasticsearch/xpack/relevancesearch/query/RelevanceMatchQueryRewriter.java

```diff
@@ -35,96 +37,182 @@
+            QueryFieldsResolver queryFieldsResolver
+        ) {
+            super(clusterService, settingsService);
+            this.queryFieldsResolver = queryFieldsResolver;
+        }
+
+        @Override
+        public QueryBuilder rewriteQuery(
+            QueryBuilder baseQuery,
+            RelevanceMatchQueryBuilder relevanceMatchQuery,
+            SearchExecutionContext context
+        ) {
+            return rewriteQuery(relevanceMatchQuery, context);
+        }
+
+        public QueryBuilder rewriteQuery(RelevanceMatchQueryBuilder relevanceMatchQuery, SearchExecutionContext context) {
+            Map<String, Float> fieldsAndBoosts = retrieveFieldsAndBoosts(relevanceMatchQuery, context);
+            return new CombinedFieldsQueryBuilder(relevanceMatchQuery.getQuery(), fieldsAndBoosts);
+        }
+
+        @Override
+        protected String getSettingsId(SearchEngine searchEngine) {
+            return searchEngine.getRelevanceSettingsId();
+        }
+
+        @Override
+        protected String getSettingsId(RelevanceMatchQueryBuilder relevanceMatchQuery) {
+            return relevanceMatchQuery.getRelevanceSettingsId();
+        }
+
+        private Map<String, Float> retrieveFieldsAndBoosts(RelevanceMatchQueryBuilder relevanceMatchQuery, SearchExecutionContext context) {
+            RelevanceSettings relevanceSettings = this.getSettings(relevanceMatchQuery, context);
+            if (relevanceSettings != null) {
+                return relevanceSettings.getQueryConfiguration().getFieldsAndBoosts();
             }
-        } else {
+
             Collection<String> fields = queryFieldsResolver.getQueryFields(context);
             if (fields.isEmpty()) {
                 throw new IllegalArgumentException("[relevance_match] query cannot find text fields in the index");
             }
-            fieldsAndBoosts = fields.stream().collect(Collectors.toMap(Function.identity(), (field) -> AbstractQueryBuilder.DEFAULT_BOOST));
+            return fields.stream().collect(Collectors.toMap(Function.identity(), (field) -> AbstractQueryBuilder.DEFAULT_BOOST));
         }
-        return fieldsAndBoosts;
     }
 
-    private QueryBuilder applyCurations(QueryBuilder queryBuilder, RelevanceMatchQueryBuilder relevanceMatchQueryBuilder) {
-        String curationsSettingsId = relevanceMatchQueryBuilder.getCurationsSettingsId();
-        if (curationsSettingsId != null) {
-            try {
-                CurationSettings curationSettings = curationsService.getSettings(curationsSettingsId);
-                final boolean conditionMatch = curationSettings.conditions().stream().anyMatch(c -> c.match(relevanceMatchQueryBuilder));
-                if (conditionMatch) {
-                    queryBuilder = applyPinnedDocs(queryBuilder, curationSettings);
-                    queryBuilder = applyExcludedDocs(queryBuilder, curationSettings);
+    private static class CurationQueryRewriter extends AbstractQueryRewriter<CurationSettings> {
```

Comment:
I think it makes sense to separate these classes to their own class file; I think we will be expanding them in a few iterations 😉